serenity-cucumber/src/main/java/io/cucumber/junit/CucumberSerenityBaseRunner.java

```diff
@@ -0,0 +1,376 @@
+package io.cucumber.junit;
+
+import static io.cucumber.core.runtime.SynchronizedEventBus.synchronize;
+import static io.cucumber.junit.FileNameCompatibleNames.uniqueSuffix;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+import static net.thucydides.core.ThucydidesSystemProperty.SERENITY_BATCH_COUNT;
+import static net.thucydides.core.ThucydidesSystemProperty.SERENITY_BATCH_NUMBER;
+import static net.thucydides.core.ThucydidesSystemProperty.SERENITY_FORK_COUNT;
+import static net.thucydides.core.ThucydidesSystemProperty.SERENITY_FORK_NUMBER;
+import io.cucumber.core.eventbus.EventBus;
+import io.cucumber.core.feature.FeatureParser;
+import io.cucumber.core.filter.Filters;
+import io.cucumber.core.gherkin.Feature;
+import io.cucumber.core.gherkin.Pickle;
+import io.cucumber.core.options.*;
+import io.cucumber.core.plugin.PluginFactory;
+import io.cucumber.core.plugin.Plugins;
+import io.cucumber.core.plugin.SerenityReporter;
+import io.cucumber.core.resource.ClassLoaders;
+import io.cucumber.core.runtime.*;
+import io.cucumber.plugin.Plugin;
+import io.cucumber.tagexpressions.Expression;
+import java.net.URI;
+import java.time.Clock;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import lombok.extern.slf4j.Slf4j;
+import net.serenitybdd.cucumber.SerenityOptions;
+import net.serenitybdd.cucumber.suiteslicing.CucumberSuiteSlicer;
+import net.serenitybdd.cucumber.suiteslicing.ScenarioFilter;
+import net.serenitybdd.cucumber.suiteslicing.TestStatistics;
+import net.serenitybdd.cucumber.suiteslicing.WeightedCucumberScenarios;
+import net.serenitybdd.cucumber.util.PathUtils;
+import net.serenitybdd.cucumber.util.Splitter;
+import net.thucydides.core.ThucydidesSystemProperty;
+import net.thucydides.core.environment.SystemEnvironmentVariables;
+import net.thucydides.core.steps.StepEventBus;
+import net.thucydides.core.util.EnvironmentVariables;
+import org.junit.runner.Description;
+import org.junit.runner.manipulation.NoTestsRemainException;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.ParentRunner;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.RunnerScheduler;
+import org.junit.runners.model.Statement;
+
+@Slf4j
+public class CucumberSerenityBaseRunner extends ParentRunner<ParentRunner<?>> {
+
+
+    private EventBus bus;
+
+    private Plugins plugins;
+
+    private List<ParentRunner<?>> children;
+
+    private List<Feature> features;
+    private CucumberExecutionContext context;
+
+    protected boolean multiThreadingAssumed = false;
+
+
+    static ThreadLocal<RuntimeOptions> RUNTIME_OPTIONS = new ThreadLocal<>(); // NOSONAR
```

Comment:
looks like this comment is not needed here

Reply:
This is needed as Sonar code analysis is biting us here with a false positive warning